Cannot invoke "java.util.jar.Manifest.getMainAttributes()" because the return value of "java.util.jar.JarFile.getManifest()" is null

`jbang -m com.oracle.graal.python.shell.GraalPythonMain org.graalvm.python:python-launcher:23.1.0`

and 
`jbang -m com.oracle.graal.python.shell.GraalPythonMain org.graalvm.python:python-launcher:23.1.0`

both fail with 

```
[jbang] [0:246] [ERROR] Cannot invoke "java.util.jar.Manifest.getMainAttributes()" because the return value of "java.util.jar.JarFile.getManifest()" is null
java.lang.NullPointerException: Cannot invoke "java.util.jar.Manifest.getMainAttributes()" because the return value of "java.util.jar.JarFile.getManifest()" is null
	at dev.jbang.source.ProjectBuilder.importJarMetadata(ProjectBuilder.java:368)
	at dev.jbang.source.ProjectBuilder.createJarProject(ProjectBuilder.java:281)
	at dev.jbang.source.ProjectBuilder.build(ProjectBuilder.java:265)
	at dev.jbang.source.ProjectBuilder.build(ProjectBuilder.java:219)
	at dev.jbang.cli.Run.doCall(Run.java:73)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:145)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:21)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
	at dev.jbang.cli.JBang$3.handle(JBang.java:148)
	at dev.jbang.cli.JBang$3.handle(JBang.java:143)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at dev.jbang.Main.main(Main.java:14)
```

class has just a module-info, no manifest.

--- a/src/main/java/dev/jbang/source/ProjectBuilder.java
+++ b/src/main/java/dev/jbang/source/ProjectBuilder.java
@@ -365,9 +365,16 @@ public class ProjectBuilder {
 					prj.setModuleName(moduleName);
 				}
 
-				Attributes attrs = jf.getManifest().getMainAttributes();
-				if (attrs.containsKey(Attributes.Name.MAIN_CLASS)) {
-					prj.setMainClass(attrs.getValue(Attributes.Name.MAIN_CLASS));
+				if (jf.getManifest() != null) {
+					Attributes attrs = jf.getManifest().getMainAttributes();
+					if (attrs.containsKey(Attributes.Name.MAIN_CLASS)) {
+						prj.setMainClass(attrs.getValue(Attributes.Name.MAIN_CLASS));
+					}
+					String ver = attrs.getValue(JarBuildStep.ATTR_BUILD_JDK);
+					if (ver != null) {
+						// buildJdk = JavaUtil.parseJavaVersion(ver);
+						prj.setJavaVersion(JavaUtil.parseJavaVersion(ver) + "+");
+					}
 				}
 
 				Optional<JarEntry> pom = jf.stream().filter(e -> e.getName().endsWith("/pom.xml")).findFirst();
@@ -390,11 +397,6 @@ public class ProjectBuilder {
 					}
 				}
 
-				String ver = attrs.getValue(JarBuildStep.ATTR_BUILD_JDK);
-				if (ver != null) {
-					// buildJdk = JavaUtil.parseJavaVersion(ver);
-					prj.setJavaVersion(JavaUtil.parseJavaVersion(ver) + "+");
-				}
 			} catch (IOException e) {
 				Util.warnMsg("Problem reading manifest from " + jar);
 			}

--- a/src/test/java/dev/jbang/cli/TestRun.java
+++ b/src/test/java/dev/jbang/cli/TestRun.java
@@ -2406,4 +2406,24 @@ public class TestRun extends BaseTest {
 		assertThat(result.err, containsString("foo%{" + arg + "}bar"));
 		assertThat(result.err, not(containsString("foo%%{" + arg + "}bar")));
 	}
+
+	@Test
+	void testHelloWorldGAVWithModulesButNoManifest() throws IOException {
+		environmentVariables.clear("JAVA_HOME");
+		String jar = "org.graalvm.python:python-launcher:23.1.0";
+
+		CommandLine.ParseResult pr = JBang.getCommandLine().parseArgs("run", jar);
+		Run run = (Run) pr.subcommand().commandSpec().userObject();
+
+		ProjectBuilder pb = run.createProjectBuilderForRun();
+		Project code = pb.build(jar);
+
+		// assertThat(code.getResourceRef().getFile().toString(),
+		// matchesPattern(".*jbang_tests_maven.*codegen-4.6.3.jar"));
+
+		ExitException e = Assertions.assertThrows(ExitException.class,
+				() -> run.updateGeneratorForRun(CmdGenerator.builder(code)).build().generate());
+
+		assertThat(e.getMessage(), startsWith("no main class"));
+	}
 }

